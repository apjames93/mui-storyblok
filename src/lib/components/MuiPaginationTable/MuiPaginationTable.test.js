import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiPaginationTable from './MuiPaginationTable';

const row = {
  _editable: '<!--#storyblok#{"name": "MuiPaginationTable", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

  component: 'MuiPaginationTableRow',
  content: [
    {
      _editable: '<!--#storyblok#{"name": "MuiTableCell", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

      component: 'MuiTableCell',
      content: [
        {
          _editable: '<!--#storyblok#{"name": "MuiTypography", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

          component: 'MuiTypography',
          content: [
            {
              _editable: '<!--#storyblok#{"name": "MuiText", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

              component: 'MuiText',
              text: 'wooo',
            },
          ],
        },
      ],
    },
  ],
};

const rows = [];
do {
  // eslint-disable-next-line no-plusplus
  rows.push(row);
} while (rows.length < 12);

function setup(tBody = rows) {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiPaginationTable", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

    tHead: [
      {
        content: [
          {
            _editable: '<!--#storyblok#{"name": "MuiPaginationTableRow", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

            component: 'MuiPaginationTableRow',
            content: [
              {
                _editable: '<!--#storyblok#{"name": "MuiTableCell", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

                component: 'MuiTableCell',
                content: [
                  {
                    _editable: '<!--#storyblok#{"name": "MuiTypography", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

                    component: 'MuiTypography',
                    content: [
                      {
                        _editable: '<!--#storyblok#{"name": "MuiText", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',

                        component: 'MuiText',
                        text: 'wooo',
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
    tBody: [
      {
        _editable: '<!--#storyblok#{"name": "MuiPaginationTableBody", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiPaginationTableBody',
        content: tBody,
      },
    ],
  };
  const comp = mount(<MuiPaginationTable {...props} />);
  return { comp, props };
}


describe('<MuiPaginationTable />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiPaginationTable {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiPaginationTable', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('should display the correct amount of rows after next/previous button clicked', () => {
    const { comp } = setup();
    const btn = comp.find('[data-testid="nextButton"]');
    const previousBtn = comp.find('[data-testid="previousButton"]');
    btn.first().simulate('click');
    const doesIncludeNext = comp.text().includes('11-12 of 12');
    expect(doesIncludeNext).toEqual(true);
    previousBtn.first().simulate('click');
    const doesIncludePrevious = comp.text().includes('1-10 of 12');
    expect(doesIncludePrevious).toEqual(true);
  });

  it('should display the correct amount of rows after first/last button clicked', () => {
    const newRows = [];
    do {
      newRows.push(row);
    } while (newRows.length < 50);

    const { comp } = setup(newRows);
    const lastBtn = comp.find('[data-testid="lastButton"]');
    const firstBtn = comp.find('[data-testid="firstButton"]');
    lastBtn.first().simulate('click');
    const doesIncludeLast = comp.text().includes('41-50 of 50');
    expect(doesIncludeLast).toEqual(true);
    firstBtn.first().simulate('click');
    const doesIncludeFirst = comp.text().includes('1-10 of 50');
    expect(doesIncludeFirst).toEqual(true);
  });

  it('should handleChangeRows and set the number or rows per page', () => {
    const { comp } = setup();
    const changeRows = comp.find('MuiPaginationTableFooter').first().prop('handleChangeRows');
    changeRows({ target: { value: 10 } });
    const page = comp.find('MuiPaginationTableFooter').first().prop('page');
    expect(page).toEqual(0);
    const rowsPerPage = comp.find('WithStyles(ForwardRef(TablePagination))').first().prop('rowsPerPage');
    expect(rowsPerPage).toEqual(10);
  });
});
