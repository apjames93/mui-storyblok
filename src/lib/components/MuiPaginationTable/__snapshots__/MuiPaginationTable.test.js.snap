// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiPaginationTable /> snapshot 1`] = `
<table
  className="MuiTable-root makeStyles-root-101"
  role={null}
>
  <thead
    className="MuiTableHead-root makeStyles-root-102"
    role={null}
  >
    <tr
      className="MuiTableRow-root makeStyles-root-103 MuiTableRow-head"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head makeStyles-root-104"
        scope="col"
      >
        <div
          className="MuiTypography-root makeStyles-root-105 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-106 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root makeStyles-root-107"
    role={null}
  >
    <tr
      className="MuiTableRow-root makeStyles-root-108"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-109"
      >
        <div
          className="MuiTypography-root makeStyles-root-110 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-111 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-112"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-113"
      >
        <div
          className="MuiTypography-root makeStyles-root-114 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-115 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-116"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-117"
      >
        <div
          className="MuiTypography-root makeStyles-root-118 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-119 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-120"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-121"
      >
        <div
          className="MuiTypography-root makeStyles-root-122 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-123 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-124"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-125"
      >
        <div
          className="MuiTypography-root makeStyles-root-126 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-127 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-128"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-129"
      >
        <div
          className="MuiTypography-root makeStyles-root-130 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-131 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-132"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-133"
      >
        <div
          className="MuiTypography-root makeStyles-root-134 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-135 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-136"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-137"
      >
        <div
          className="MuiTypography-root makeStyles-root-138 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-139 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-140"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-141"
      >
        <div
          className="MuiTypography-root makeStyles-root-142 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-143 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-144"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-145"
      >
        <div
          className="MuiTypography-root makeStyles-root-146 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-147 MuiTypography-body1"
          >
             undefined 
          </span>
        </div>
      </td>
    </tr>
  </tbody>
  <tfoot
    className="MuiTableFooter-root makeStyles-root-148"
    role={null}
  >
    <tr
      className="MuiTableRow-root MuiTableRow-footer"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root makeStyles-root-149"
        colSpan={1000}
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            className="MuiTablePagination-spacer"
          />
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            Rows per page:
          </p>
          <div
            className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              10
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              tabIndex={-1}
              value={10}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            0-0 of 0
          </p>
          <div
            className="makeStyles-root-150"
          >
            <button
              aria-label="first page"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              data-testid="firstButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                  />
                  <path
                    d="M24 24H0V0h24v24z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="previous page"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              data-testid="previousButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="next page"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              data-testid="nextButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="last page"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              data-testid="lastButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tfoot>
</table>
`;
